packages/twenty-server/src/engine/core-modules/throttler/throttler.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@nestjs/common';
+
+import {
+  ThrottlerException,
+  ThrottlerExceptionCode,
+} from 'src/engine/core-modules/throttler/throttler.exception';
+import { CacheStorageService } from 'src/engine/integrations/cache-storage/cache-storage.service';
+import { InjectCacheStorage } from 'src/engine/integrations/cache-storage/decorators/cache-storage.decorator';
+import { CacheStorageNamespace } from 'src/engine/integrations/cache-storage/types/cache-storage-namespace.enum';
+
+@Injectable()
+export class ThrottlerService {
+  constructor(
+    @InjectCacheStorage(CacheStorageNamespace.EngineWorkspace)
+    private readonly cacheStorage: CacheStorageService,
+  ) {}
+
+  async throttle(key: string, limit: number, ttl: number): Promise<void> {
+    const cached = await this.cacheStorage.get<number>(key);
+
+    if (cached && cached >= limit) {
+      throw new ThrottlerException(
+        'Too many requests',
+        ThrottlerExceptionCode.TOO_MANY_REQUESTS,
+      );
+    }
+
+    if (cached) {
```

Comment:
I don't think this is needed? 
```typescript
await this.cacheStorage.set(key, currentCount + 1, ttl);
```
if currentCount is null it should still put 1

Reply:
nevermind, I forgot this would return undefined and not null! I've suggested a different change above, up to you